### ISSUE TITLE
Cannot fine tune with ilsvrc15 SSD 300x300

### ISSUE THREAD

**Issue by leosteets** [weiliu89/caffe#120]
I'm successfully trained my network from scratch changing the ssd_pascal.py to my data and configs. But when i'm trying to fune-tune it with ilsvrc15 I got an error, as you can see below.

```
I0817 00:43:19.937486 164801 caffe.cpp:129] Finetuning from /root/Leo/caffe/models/VGGNet/ilsvrc15/SSD_300x300/VGG_ilsvrc15_SSD_300x300_iter_480000.caffemodel
F0817 00:43:21.242597 164801 net.cpp:765] Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_conf'; shape mismatch.  Source param shape is 603 512 3 3 (2778624); target param shape is 99 512 3 3 (456192). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
*** Check failure stack trace: ***
    @     0x7fb8c3dbcdaa  (unknown)
    @     0x7fb8c3dbcce4  (unknown)
    @     0x7fb8c3dbc6e6  (unknown)
    @     0x7fb8c3dbf687  (unknown)
    @     0x7fb8c463ba57  caffe::Net<>::CopyTrainedLayersFrom()
    @     0x7fb8c4642212  caffe::Net<>::CopyTrainedLayersFromBinaryProto()
    @     0x7fb8c4642276  caffe::Net<>::CopyTrainedLayersFrom()
    @           0x407ed4  CopyLayers()
    @           0x4086cd  train()
    @           0x4060fc  main
    @     0x7fb8c30caf45  (unknown)
    @           0x406831  (unknown)
    @              (nil)  (unknown)

```

What should I do? Should I reshape the conv4_3_norm_mbox_conf layer?

Why is it different?

Cheers.

**Comment by weiliu89** [comment]
You have 32 + 1 classes for your dataset, while ILSVRC has 200 + 1 classes. If there is overlap between your classes and ILSVRC classes, you can extract those corresponding confidence predictions. You can keep the location predictions as those are shared by all classes.

**Comment by leejiajun** [comment]
@weiliu89 
Hi, I train SSD with ilsvrc2014, but I got an error.

> I0817 15:19:18.042752  7242 layer_factory.hpp:77] Creating layer detection_eval
I0817 15:19:18.042773  7242 net.cpp:91] Creating Layer detection_eval
I0817 15:19:18.042778  7242 net.cpp:425] detection_eval <- detection_out
I0817 15:19:18.042784  7242 net.cpp:425] detection_eval <- label
I0817 15:19:18.042793  7242 net.cpp:399] detection_eval -> detection_eval
F0817 15:19:18.046648  7242 detection_evaluate_layer.cpp:52] Check failed: bottom[1]->width() == 8 (1 vs. 8)
*** Check failure stack trace: ***
    @     0x7fedb8c94daa  (unknown)
    @     0x7fedb8c94ce4  (unknown)
    @     0x7fedb8c946e6  (unknown)
    @     0x7fedb8c97687  (unknown)
    @     0x7fedb930f88d  caffe::DetectionEvaluateLayer<>::Reshape()
    @     0x7fedb947f6fc  caffe::Net<>::Init()
    @     0x7fedb9480585  caffe::Net<>::Net()
    @     0x7fedb94e5b6a  caffe::Solver<>::InitTestNets()
    @     0x7fedb94e633d  caffe::Solver<>::Init()
    @     0x7fedb94e662a  caffe::Solver<>::Solver()
    @     0x7fedb9456a33  caffe::Creator_SGDSolver<>()
    @           0x41005e  caffe::SolverRegistry<>::CreateSolver()
    @           0x409490  train()
    @           0x406f2c  main
    @     0x7fedb7fa2f45  (unknown)
    @           0x407661  (unknown)
    @              (nil)  (unknown)
                                           
 What should I do? 

Thank you.

**Comment by shawn-tian** [comment]
i think it is probably your lmdb for validation set is not properly
formatted and written.

**Comment by leejiajun** [comment]
@shawn-tian I use val1 and val2 provided by @weiliu89.  And I also fix images.

**Comment by weiliu89** [comment]
It looks like you didn't output label correctly from the annotated_data_layer. You should debug what is going on. I am guessing you didn't store the annotation correctly.

**Comment by leosteets** [comment]
Thanks for yout reply.

No, I don't have any overlap between the ilsvrc15 classes. Is a total different and specific objects.

@weiliu89 so, in bottom line I cannot fine-tune using any other pretrained weights since the weights consider the class probability location of each class right?

Just one more thing, if I can't use other pratrained net, how can I use the "reduced fc" model to train my own net? I got lost here. And there is any difference between using the "reduced fc" and training from scratch?

**Comment by weiliu89** [comment]
If you try to fine-tune from an existing SSD model (e.g. SSD trained from ILSVRC DET or COCO), then you need to change the name for your confidence prediction layers (to not load the confidence predictions from an existing SSD model), and set a smaller learning rate.

If you try to fine-tune from reduced VGG model (trained on ILSVRC classification dataset), then you can follow ssd_pascal.py to make changes accordingly.

These are some general practices how you fine-tune models. You can probably find other references on how to do fine-tuning (e.g. the official Caffe site)

**Comment by leosteets** [comment]
Thank you so much, this was really helpfull.

Cheers.
